pkg/util/pki/csr_test.go

```diff
@@ -378,22 +378,14 @@ func TestGenerateCSR(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	asn1ExtKeyUsage, err := asn1.Marshal([]asn1.ObjectIdentifier{})
-	if err != nil {
-		t.Fatal(err)
-	}
 	defaultExtraExtensions := []pkix.Extension{
 		{
 			Id:    OIDExtensionKeyUsage,
 			Value: asn1KeyUsage,
 		},
-		{
-			Id:    OIDExtensionExtendedKeyUsage,
-			Value: asn1ExtKeyUsage,
-		},
 	}
 
-	asn1ExtKeyUsage, err = asn1.Marshal([]asn1.ObjectIdentifier{oidExtKeyUsageIPSECEndSystem})
+	asn1ExtKeyUsage, err := asn1.Marshal([]asn1.ObjectIdentifier{oidExtKeyUsageIPSECEndSystem})
```

Comment:
As mentioned above, I think it'd be nice to a separate function called `extraUsagesForCertificate` which we can unit test separately.
Then we'd be able to write a specifc test case for the situation where no extended usages are supplied.